magma_access_gateway_installer/__init__.py

```diff
@@ -69,18 +72,73 @@ def cli_arguments_parser(cli_arguments):
         "--ip-address",
         dest="ip_address",
         required=False,
-        help="Statically allocated SGi interface IP address. Example: 1.1.1.1/24",
+        help="Statically allocated SGi interface IP address. Example: 1.1.1.1/24.",
     )
     cli_options.add_argument(
         "--gw-ip-address",
         dest="gw_ip_address",
         required=False,
-        help="Upstream router IP for SGi interface. Example: 1.1.1.200",
+        help="Upstream router IP for SGi interface. Example: 1.1.1.200.",
+    )
+    cli_options.add_argument(
+        "--skip-networking",
+        dest="skip_networking",
+        action="store_true",
+        required=False,
+        help="If used, network configuration part of the installation process will be skipped. "
+        "In this case operator is responsible for providing expected network configuration.",
+    )
+    cli_options.add_argument(
+        "--dns",
+        dest="dns",
+        nargs="+",
+        required=False,
+        default=["8.8.8.8", "208.67.222.222"],
+        help="Space separated list of DNS IP addresses. Example: --dns 1.2.3.4 5.6.7.8.",
+    )
+    cli_options.add_argument(
+        "--sgi",
+        dest="sgi",
+        required=False,
+        help="Defines which interface should be used as SGi interface.",
+    )
+    cli_options.add_argument(
+        "--s1",
+        dest="s1",
+        required=False,
+        help="Defines which interface should be used as S1 interface.",
     )
     return cli_options.parse_args(cli_arguments)
 
 
-def validate_args(args):
+def validate_args(args, network_interfaces):
+    validate_static_ip_allocation(args)
+    validate_arbitrary_dns(args)
+    validate_custom_sgi_and_s1_interfaces(args, network_interfaces)
+
+
+def validate_custom_sgi_and_s1_interfaces(args, network_interfaces):
+    if args.sgi and not args.s1:
+        raise ValueError("S1 interface not specified! Exiting...")
+    elif not args.sgi and args.s1:
+        raise ValueError("SGi interface not specified! Exiting...")
+    elif args.sgi and args.s1:
+        if not all([interface in network_interfaces for interface in [args.sgi, args.s1]]):
+            raise ValueError("One or both specified SGi/S1 interfaces do not exist! Exiting...")
+
+
+def validate_arbitrary_dns(args):
+    if not args.dns:
+        raise ValueError("--dns flag specified, but no addresses given! Exiting...")
+    else:
+        for dns_ip in args.dns:
+            try:
+                ip_address(dns_ip)
+            except Exception as e:
+                raise e
```

Comment:
Do we need this `else`?
```suggestion
    for dns_ip in args.dns:
        try:
            ip_address(dns_ip)
        except Exception as e:
            raise e
```

Reply:
Not necessarily. The effect will be exactly the same. I though traditional `if - else` will be easier to read.